src/babel.ts

```diff
@@ -1,34 +1,110 @@
-import { Visitor, types as t } from '@babel/core';
-
-const babelPlugin = (): { visitor: Visitor } => ({
-  visitor: {
-    CallExpression(path) {
-      // prevents infinitely loops
-      if (t.isConditionalExpression(path.parent)) return;
-
-      if (!t.isIdentifier(path.node.callee)) return;
-      if (path.node.callee.name !== 'createDataHook') return;
-
-      const [firstArgument] = path.node.arguments;
-      if (!firstArgument) return;
-      if (!t.isStringLiteral(firstArgument)) return;
-      const dataKey = firstArgument.value;
-
-      path.replaceWith(
-        t.conditionalExpression(
-          t.binaryExpression(
-            '!==',
-            t.unaryExpression('typeof', t.identifier('window')),
-            t.stringLiteral('undefined')
-          ),
-          t.callExpression(t.identifier('createDataHook'), [
-            t.stringLiteral(dataKey),
-          ]),
-          path.node
-        )
-      );
+import { Visitor, types as t, NodePath } from '@babel/core';
+
+const PACKAGE_NAME = 'next-data-hooks';
+
+/** returns an identifier for a function from ./babel-utils.
+ * automatically imports it if it is not already imported.
+*/
+function autoImportUtility(path: NodePath<any>, util: string) {
+  const program = path.isProgram()
+    ? path
+    : path.findParent(p => p.isProgram()) as NodePath<t.Program>;
+  if (!program) throw new Error("Cannot find Program node.");
+
+  // find an import statement to the utility module
+  let utilImport = program.get('body').find(x =>
+    x.isImportDeclaration()
+    && x.get('source').node.value === PACKAGE_NAME
+    && !x.get('specifiers').some(s => s.isImportNamespaceSpecifier())
+  ) as NodePath<t.ImportDeclaration>;
```

Comment:
I'm getting some ts warnings here. let me know if you're up to fighting the TS compiler otherwise I can try to tame it lol

![image](https://user-images.githubusercontent.com/10551026/114506015-37333280-9bff-11eb-839b-7976e69ac1a1.png)